src/types.ts

```diff
@@ -163,6 +225,8 @@ export interface Check extends BaseObject {
 
   // Link to probes
   probes: number[];
+  id?: number;
+  tenantId?: number;
 }
 
 export interface Settings {
```

Comment:
are these mutually exclusive, or is it possible for example of both `http` and `ping` properties to be on the `Settings` object?

Reply:
They are mutually exclusive. There must be a better way to type this situation, right?

Reply:
I think there's also a way to make `ping`, `http`, etc properties exclusive, but type description is too gnarly to recommend

Reply:
You don't need the additional `_type`. You can just check for a property that's exclusive to one of the types (so in this case `'http' in settings`) and TS will narrow it: https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8UDeBDAXFAzsATgSwDsBzAXwG4AoUSKAIQWQCMNt9jyrxpaB7YACwZwAPnUoUAZgFcCAY2B4eBKPwC2AChQ4iGXgICUyClCh4J6gOQoLp5VqKGkxk1FlLMPADYQAdJ55Emtr6lCYkUBCemNBOLq7uXr7+gfYhziQUzm4EHt5+AUEOlBlAA

Reply:
Thanks y'all, I'll ruminate on a good way to proceed here